File: 백준/Silver/10828. 스택/스택.py
import sys
n=int(sys.stdin.readline())
s=[]
for _ in range(n):
    m=sys.stdin.readline().split()
    if m[0]=='push':
      s.append(m[1])
    elif m[0]=='top':
        print(-1 if len(s)==0 else s[-1])
    elif m[0]=='size':
        print(len(s))
    elif m[0]=='empty':
        print(1 if len(s)==0 else 0)
    elif m[0]=='pop':
        if len(s)==0: 
            print(-1)
        else:
            print(s.pop())   
            

File: 백준/Silver/10828. 스택/스택.cc
#include <iostream>
#include <stack>
#include <string>

using namespace std;

typedef long long ll;
ll n, s2;
string s1;
int main()
{
  ios::sync_with_stdio(false);
  cin.tie(NULL);
  stack<ll> st;
  cin >> n;
  for (int i = 0; i < n; i++) {
    cin >> s1;
    if (s1 == "push") {
      cin >> s2;
      st.push(s2);
    } else if (s1 == "top") {
      if (st.empty())
        cout << -1 << "\n";
      else
        cout << st.top() << "\n";
    } else if (s1 == "pop") {
      if (st.empty())
        cout << -1 << "\n";
      else {
        cout << st.top() << "\n";
        st.pop();
      }  
    } else if (s1 == "size") {
      cout << st.size() << "\n";
    } else if (s1 == "empty") {
      if (st.empty())
        cout << "1\n";
      else
        cout << "0\n";
    }
  }
}